base/compiler/typelattice.jl

```diff
@@ -294,12 +312,12 @@ The non-strict partial order over the type inference lattice.
             return a.instance === b.val
         end
         return false
-    elseif isa(a, PartialTypeVar) && b === TypeVar
-        return true
-    elseif isa(a, Type) && isa(b, Type)
-        return a <: b
-    else # handle this conservatively in the remaining cases
+    elseif isa(a, PartialTypeVar)
+        return b === TypeVar || a === b
```

Comment:
Is it legal to compare these with `===`? That is not a lattice-preserving operation for the TypeVar object inside.

Reply:
That is unchanged in this PR, the condition just moved around. I agree that it might not be sound though.

Reply:
Ah, gotcha. It was hard to see that in the default unified diff; easier in the split diff.